HPX 1.7.0-rc1 fails to build icw APEX + OTF2

## Expected Behavior

Build succeeds.

## Actual Behavior

Build fails with these messages (full log attached):

```
apex-src/src/apex/otf2_listener.cpp:1402:59: error: ‘gather_there’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_there’?
apex-src/src/apex/otf2_listener.cpp:1412:28: error: ‘gather_here’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_here’?
apex-src/src/apex/otf2_listener.cpp:1455:59: error: ‘gather_there’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_there’? apex-src/src/apex/otf2_listener.cpp:1464:32: error: ‘gather_here’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_here’?
apex-src/src/apex/otf2_listener.cpp:1526:32: error: ‘broadcast_from’ is not a member of ‘hpx::lcos’; did you mean ‘broadcast_apply’?
apex-src/src/apex/otf2_listener.cpp:1526:58: error: expected primary-expression before ‘>’ token               
apex-src/src/apex/otf2_listener.cpp:1531:32: error: ‘broadcast_to’ is not a member of ‘hpx::lcos’; did you mean ‘broadcast’?
apex-src/src/apex/otf2_listener.cpp:1585:59: error: ‘gather_there’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_there’?
apex-src/src/apex/otf2_listener.cpp:1594:32: error: ‘gather_here’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_here’?
apex-src/src/apex/otf2_listener.cpp:1662:32: error: ‘broadcast_from’ is not a member of ‘hpx::lcos’; did you mean ‘broadcast_apply’?
apex-src/src/apex/otf2_listener.cpp:1662:58: error: expected primary-expression before ‘>’ token               
apex-src/src/apex/otf2_listener.cpp:1667:32: error: ‘broadcast_to’ is not a member of ‘hpx::lcos’; did you mean ‘broadcast’?
apex-src/src/apex/otf2_listener.cpp:1714:59: error: ‘gather_there’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_there’?
apex-src/src/apex/otf2_listener.cpp:1724:28: error: ‘gather_here’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_here’?
```

## Steps to Reproduce the Problem

  1. Configure HPX with `-DHPX_WITH_APEX:BOOL=ON -DAPEX_WITH_OTF2:BOOL=ON -DHELLO_HPX_BUILD_OTF2:BOOL=ON -DHPX_WITH_EXAMPLES:BOOL=ON`
  
I test HPX builds using this project: https://github.com/kordejong/hello_hpx The build that fails is the one in https://github.com/kordejong/hello_hpx/blob/b8f53591c9b6fcfcc6e5ce887bd48c23ffe2c77d/rel_with_deb_info_build.sh#L12


## Specifications

  - HPX Version: 1.7.0-rc1
  - Platform (compiler, OS): gcc-9.3.0, Linux

[build_log.txt](https://github.com/STEllAR-GROUP/hpx/files/6662989/build_log.txt)

--- a/libs/full/collectives/docs/index.rst
+++ b/libs/full/collectives/docs/index.rst
@@ -1,5 +1,5 @@
 ..
-    Copyright (c) 2019 The STE||AR-Group
+    Copyright (c) 2019-2021 The STE||AR-Group
 
     SPDX-License-Identifier: BSL-1.0
     Distributed under the Boost Software License, Version 1.0. (See accompanying
@@ -15,20 +15,38 @@ The collectives module exposes a set of distributed collective operations. Those
 can be used to exchange data between participating sites in a coordinated way.
 At this point the module exposes the following collective primitives:
 
-* :cpp:class:`hpx::collectives::all_reduce`: performs a reduction on data from
+* :cpp:func:`hpx::collectives::all_gather`: receives a set of values from all
+  participating sites.
+* :cpp:func:`hpx::collectives::all_reduce`: performs a reduction on data from
   each participating site to each participating site.
-* :cpp:class:`hpx::collectives::all_to_all`: each participating site provides its
+* :cpp:func:`hpx::collectives::all_to_all`: each participating site provides its
   element of the data to collect while all participating sites receive the data
   from every other site.
-* :cpp:class:`hpx::lcos::barrier`: distributed barrier.
+* :cpp:func:`hpx::collectives::broadcast_to` and
+  :cpp:func:`hpx::collectives::broadcast_from`: performs a broadcast operation
+  from a root site to all participating sites.
+* :cpp:func:hpx::collectives::exclusive_scan: performs an exclusive scan operation
+  on a set of values received from all call sites operating on the given base name.
+* :cpp:func:`hpx::collectives::gather_here` and
+  :cpp:func:`hpx::collectives::gather_there`: gathers values from all
+  participating sites.
+* :cpp:func:hpx::collectives::inclusive_scan: performs an inclusive scan operation
+  on a set of values received from all call sites operating on the given base name.
+* :cpp:func:`hpx::collectives::reduce_here` and
+  :cpp:func:`hpx::collectives::reduce_there`: performs a reduction on data from each
+  participating site to a root site.
+* :cpp:func:`hpx::collectives::scatter_to` and
+  :cpp:func:`hpx::collectives::scatter_from`: receives an element of a set of values
+  operating on the given base name.
+
 * :cpp:func:`hpx::lcos::broadcast`: performs a given action on all given global
   identifiers.
+* :cpp:class:`hpx::lcos::barrier`: distributed barrier.
 * :cpp:func:`hpx::lcos::fold`: performs a fold with a given action on all given
   global identifiers.
-* :cpp:func:`hpx::lcos::gather`: gathers values from all participating sites.
 * :cpp:class:`hpx::lcos::latch`: distributed latch.
 * :cpp:func:`hpx::lcos::reduce`: performs a reduction on data from each
-  participating site to a root site.
+  given global identifiers.
 * :cpp:class:`hpx::lcos::spmd_block`: performs the same operation on a local
   image while providing handles to the other images.
 

--- a/libs/full/collectives/include/hpx/collectives/all_gather.hpp
+++ b/libs/full/collectives/include/hpx/collectives/all_gather.hpp
@@ -203,6 +203,86 @@ namespace hpx { namespace collectives {
 }}    // namespace hpx::collectives
 
 ////////////////////////////////////////////////////////////////////////////////
+// compatibility functions
+namespace hpx { namespace lcos {
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_gather is deprecated, use hpx::collectives::all_gather "
+        "instead")
+    hpx::future<std::vector<std::decay_t<T>>> all_gather(char const* basename,
+        T&& local_result, std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return hpx::collectives::all_gather(basename,
+            std::forward<T>(local_result),
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation),
+            hpx::collectives::root_site_arg(root_site));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_gather is deprecated, use hpx::collectives::all_gather "
+        "instead")
+    hpx::future<std::vector<T>> all_gather(char const* basename,
+        hpx::future<T>&& local_result, std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return local_result.then([=](hpx::future<T>&& f) {
+            return hpx::collectives::all_gather(basename, f.get(),
+                hpx::collectives::num_sites_arg(num_sites),
+                hpx::collectives::this_site_arg(this_site),
+                hpx::collectives::generation_arg(generation),
+                hpx::collectives::root_site_arg(root_site));
+        });
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_gather is deprecated, use hpx::collectives::all_gather "
+        "instead")
+    hpx::future<std::vector<std::decay_t<T>>> all_gather(
+        hpx::collectives::communicator comm, T&& local_result,
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::all_gather(std::move(comm),
+            std::forward<T>(local_result),
+            hpx::collectives::this_site_arg(this_site));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_gather is deprecated, use hpx::collectives::all_gather "
+        "instead")
+    hpx::future<std::vector<T>> all_gather(hpx::collectives::communicator comm,
+        hpx::future<T>&& local_result, std::size_t this_site = std::size_t(-1))
+    {
+        return local_result.then([=](hpx::future<T>&& f) mutable {
+            hpx::collectives::all_gather(std::move(comm), f.get(),
+                hpx::collectives::this_site_arg(this_site));
+        });
+    }
+
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::create_all_gather is deprecated, use "
+        "hpx::collectives::create_communicator instead")
+    inline hpx::collectives::communicator create_all_gather(
+        char const* basename, std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::create_communicator(basename,
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation));
+    }
+}}    // namespace hpx::lcos
+
+////////////////////////////////////////////////////////////////////////////////
 #define HPX_REGISTER_ALLGATHER_DECLARATION(...) /**/
 
 ////////////////////////////////////////////////////////////////////////////////

--- a/libs/full/collectives/include/hpx/collectives/all_reduce.hpp
+++ b/libs/full/collectives/include/hpx/collectives/all_reduce.hpp
@@ -220,6 +220,91 @@ namespace hpx { namespace collectives {
 }}    // namespace hpx::collectives
 
 ////////////////////////////////////////////////////////////////////////////////
+// compatibility functions
+namespace hpx { namespace lcos {
+
+    template <typename T, typename F>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_reduce is deprecated, use hpx::collectives::all_reduce "
+        "instead")
+    hpx::future<std::decay_t<T>> all_reduce(char const* basename,
+        T&& local_result, F&& op, std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return hpx::collectives::all_reduce(basename,
+            std::forward<T>(local_result), std::forward<F>(op),
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation),
+            hpx::collectives::root_site_arg(root_site));
+    }
+
+    template <typename T, typename F>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_reduce is deprecated, use hpx::collectives::all_reduce "
+        "instead")
+    hpx::future<T> all_reduce(char const* basename,
+        hpx::future<T>&& local_result, F&& op,
+        std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return local_result.then([=](hpx::future<T>&& f) mutable {
+            // silence bogus clang-tidy error
+            return hpx::collectives::all_reduce(basename, f.get(),
+                // NOLINTNEXTLINE(bugprone-move-forwarding-reference)
+                std::move(op), hpx::collectives::num_sites_arg(num_sites),
+                hpx::collectives::this_site_arg(this_site),
+                hpx::collectives::generation_arg(generation),
+                hpx::collectives::root_site_arg(root_site));
+        });
+    }
+
+    template <typename T, typename F>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_reduce is deprecated, use hpx::collectives::all_reduce "
+        "instead")
+    hpx::future<std::decay_t<T>> all_reduce(hpx::collectives::communicator comm,
+        T&& local_result, F&& op, std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::all_reduce(std::move(comm),
+            std::forward<T>(local_result), std::forward<F>(op),
+            hpx::collectives::this_site_arg(this_site));
+    }
+
+    template <typename T, typename F>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_reduce is deprecated, use hpx::collectives::all_reduce "
+        "instead")
+    hpx::future<T> all_reduce(hpx::collectives::communicator comm,
+        hpx::future<T>&& local_result, F&& op,
+        std::size_t this_site = std::size_t(-1))
+    {
+        return local_result.then([=](hpx::future<T>&& f) mutable {
+            // silence bogus clang-tidy error
+            hpx::collectives::all_reduce(std::move(comm), f.get(),
+                // NOLINTNEXTLINE(bugprone-move-forwarding-reference)
+                std::move(op), hpx::collectives::this_site_arg(this_site));
+        });
+    }
+
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::create_all_reduce is deprecated, use "
+        "hpx::collectives::create_communicator instead")
+    inline hpx::collectives::communicator create_all_reduce(
+        char const* basename, std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::create_communicator(basename,
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation));
+    }
+}}    // namespace hpx::lcos
+
+////////////////////////////////////////////////////////////////////////////////
 #define HPX_REGISTER_ALLREDUCE_DECLARATION(...) /**/
 
 ////////////////////////////////////////////////////////////////////////////////

--- a/libs/full/collectives/include/hpx/collectives/all_to_all.hpp
+++ b/libs/full/collectives/include/hpx/collectives/all_to_all.hpp
@@ -215,6 +215,86 @@ namespace hpx { namespace collectives {
 }}    // namespace hpx::collectives
 
 ////////////////////////////////////////////////////////////////////////////////
+// compatibility functions
+namespace hpx { namespace lcos {
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_to_all is deprecated, use hpx::collectives::all_to_all "
+        "instead")
+    hpx::future<std::vector<T>> all_to_all(char const* basename,
+        std::vector<T>&& local_result, std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return hpx::collectives::all_to_all(basename, std::move(local_result),
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation),
+            hpx::collectives::root_site_arg(root_site));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_to_all is deprecated, use hpx::collectives::all_to_all "
+        "instead")
+    hpx::future<std::vector<T>> all_to_all(char const* basename,
+        hpx::future<std::vector<T>>&& local_result,
+        std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return local_result.then([=](hpx::future<T>&& f) {
+            return hpx::collectives::all_to_all(basename, f.get(),
+                hpx::collectives::num_sites_arg(num_sites),
+                hpx::collectives::this_site_arg(this_site),
+                hpx::collectives::generation_arg(generation),
+                hpx::collectives::root_site_arg(root_site));
+        });
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_to_all is deprecated, use hpx::collectives::all_to_all "
+        "instead")
+    hpx::future<std::vector<T>> all_to_all(hpx::collectives::communicator comm,
+        std::vector<T>&& local_result, std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::all_to_all(std::move(comm),
+            std::move(local_result),
+            hpx::collectives::this_site_arg(this_site));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::all_to_all is deprecated, use hpx::collectives::all_to_all "
+        "instead")
+    hpx::future<std::vector<T>> all_to_all(hpx::collectives::communicator comm,
+        hpx::future<std::vector<T>>&& local_result,
+        std::size_t this_site = std::size_t(-1))
+    {
+        return local_result.then([=](hpx::future<T>&& f) mutable {
+            hpx::collectives::all_to_all(std::move(comm), f.get(),
+                hpx::collectives::this_site_arg(this_site));
+        });
+    }
+
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::create_all_to_all is deprecated, use "
+        "hpx::collectives::create_communicator instead")
+    inline hpx::collectives::communicator create_all_to_all(
+        char const* basename, std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::create_communicator(basename,
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation));
+    }
+}}    // namespace hpx::lcos
+
+////////////////////////////////////////////////////////////////////////////////
 #define HPX_REGISTER_ALLTOALL_DECLARATION(...) /**/
 
 ////////////////////////////////////////////////////////////////////////////////

--- a/libs/full/collectives/include/hpx/collectives/broadcast.hpp
+++ b/libs/full/collectives/include/hpx/collectives/broadcast.hpp
@@ -319,6 +319,111 @@ namespace hpx { namespace collectives {
 }}    // namespace hpx::collectives
 
 ////////////////////////////////////////////////////////////////////////////////
+// compatibility functions
+namespace hpx { namespace lcos {
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::broadcast_to is deprecated, use "
+        "hpx::collectives::broadcast_to instead")
+    hpx::future<std::decay_t<T>> broadcast_to(char const* basename,
+        T&& local_result, std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return hpx::collectives::broadcast_to(basename,
+            std::forward<T>(local_result),
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation),
+            hpx::collectives::root_site_arg(root_site));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::broadcast_to is deprecated, use "
+        "hpx::collectives::broadcast_to instead")
+    hpx::future<T> broadcast_to(char const* basename,
+        hpx::future<T>&& local_result, std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return local_result.then([=](hpx::future<T>&& f) {
+            return hpx::collectives::broadcast_to(basename, f.get(),
+                hpx::collectives::num_sites_arg(num_sites),
+                hpx::collectives::this_site_arg(this_site),
+                hpx::collectives::generation_arg(generation),
+                hpx::collectives::root_site_arg(root_site));
+        });
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::broadcast is deprecated, use hpx::collectives::broadcast "
+        "instead")
+    hpx::future<typename std::decay<T>::type> broadcast_to(
+        hpx::collectives::communicator comm, T&& local_result,
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::broadcast_to(std::move(comm),
+            std::forward<T>(local_result),
+            hpx::collectives::this_site_arg(this_site));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::broadcast_to is deprecated, use "
+        "hpx::collectives::broadcast_to instead")
+    hpx::future<T> broadcast_to(hpx::collectives::communicator comm,
+        hpx::future<T>&& local_result, std::size_t this_site = std::size_t(-1))
+    {
+        return local_result.then([=](hpx::future<T>&& f) mutable {
+            hpx::collectives::broadcast_to(std::move(comm), f.get(),
+                hpx::collectives::this_site_arg(this_site));
+        });
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::broadcast_from is deprecated, use "
+        "hpx::collectives::broadcast_from instead")
+    hpx::future<T> broadcast_from(char const* basename,
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return hpx::collectives::broadcast_from<T>(basename,
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation),
+            hpx::collectives::root_site_arg(root_site));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::broadcast_from is deprecated, use "
+        "hpx::collectives::broadcast_from instead")
+    hpx::future<T> broadcast_from(hpx::collectives::communicator comm,
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::broadcast_from<T>(
+            std::move(comm), hpx::collectives::this_site_arg(this_site));
+    }
+
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::create_broadcast is deprecated, use "
+        "hpx::collectives::create_communicator instead")
+    inline hpx::collectives::communicator create_broadcast(char const* basename,
+        std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::create_communicator(basename,
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation));
+    }
+}}    // namespace hpx::lcos
+
+////////////////////////////////////////////////////////////////////////////////
 #define HPX_REGISTER_BROADCAST_DECLARATION(...) /**/
 
 ////////////////////////////////////////////////////////////////////////////////

--- a/libs/full/collectives/include/hpx/collectives/exclusive_scan.hpp
+++ b/libs/full/collectives/include/hpx/collectives/exclusive_scan.hpp
@@ -14,8 +14,8 @@ namespace hpx { namespace collectives {
 
     /// Exclusive scan a set of values from different call sites
     ///
-    /// This function receives a set of values from all call sites operating on
-    /// the given base name.
+    /// This function performs an exclusive scan operation on a set of values
+    /// received from all call sites operating on the given base name.
     ///
     /// \param  basename    The base name identifying the exclusive_scan operation
     /// \param  local_result The value to transmit to all
@@ -53,8 +53,8 @@ namespace hpx { namespace collectives {
 
     /// Exclusive scan a set of values from different call sites
     ///
-    /// This function receives a set of values from all call sites operating on
-    /// the given base name.
+    /// This function performs an exclusive scan operation on a set of values
+    /// received from all call sites operating on the given base name.
     ///
     /// \param  comm        A communicator object returned from \a create_reducer
     /// \param  local_result The value to transmit to all

--- a/libs/full/collectives/include/hpx/collectives/gather.hpp
+++ b/libs/full/collectives/include/hpx/collectives/gather.hpp
@@ -333,6 +333,141 @@ namespace hpx { namespace collectives {
     }
 }}    // namespace hpx::collectives
 
+////////////////////////////////////////////////////////////////////////////////
+// compatibility functions
+namespace hpx { namespace lcos {
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::gather_here is deprecated, use "
+        "hpx::collectives::gather_here instead")
+    hpx::future<std::vector<typename std::decay<T>::type>> gather_here(
+        char const* basename, T&& local_result,
+        std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::gather_here(basename,
+            std::forward<T>(local_result),
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::gather_here is deprecated, use "
+        "hpx::collectives::gather_here instead")
+    hpx::future<std::vector<T>> gather_here(char const* basename,
+        hpx::future<T>&& local_result, std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1))
+    {
+        return local_result.then([=](hpx::future<T>&& f) {
+            return hpx::collectives::gather_here(basename, f.get(),
+                hpx::collectives::num_sites_arg(num_sites),
+                hpx::collectives::this_site_arg(this_site),
+                hpx::collectives::generation_arg(generation));
+        });
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::gather is deprecated, use hpx::collectives::gather "
+        "instead")
+    hpx::future<std::vector<std::decay_t<T>>> gather_here(
+        hpx::collectives::communicator comm, T&& local_result,
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::gather_here(std::move(comm),
+            std::forward<T>(local_result),
+            hpx::collectives::this_site_arg(this_site));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::gather_here is deprecated, use "
+        "hpx::collectives::gather_here instead")
+    hpx::future<std::vector<T>> gather_here(hpx::collectives::communicator comm,
+        hpx::future<T>&& local_result, std::size_t this_site = std::size_t(-1))
+    {
+        return local_result.then([=](hpx::future<T>&& f) mutable {
+            hpx::collectives::gather_here(std::move(comm), f.get(),
+                hpx::collectives::this_site_arg(this_site));
+        });
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::gather_there is deprecated, use "
+        "hpx::collectives::gather_there instead")
+    hpx::future<void> gather_there(char const* basename, T&& local_result,
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return hpx::collectives::gather_there(basename,
+            std::forward<T>(local_result),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation),
+            hpx::collectives::root_site_arg(root_site));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::gather_there is deprecated, use "
+        "hpx::collectives::gather_there instead")
+    hpx::future<void> gather_there(char const* basename,
+        hpx::future<T>&& local_result, std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return local_result.then([=](hpx::future<T>&& f) {
+            return hpx::collectives::gather_there(basename, f.get(),
+                hpx::collectives::this_site_arg(this_site),
+                hpx::collectives::generation_arg(generation),
+                hpx::collectives::root_site_arg(root_site));
+        });
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::gather_there is deprecated, use "
+        "hpx::collectives::gather_there instead")
+    hpx::future<void> gather_there(hpx::collectives::communicator comm,
+        T&& local_result, std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::gather_there(std::move(comm),
+            std::forward<T>(local_result),
+            hpx::collectives::this_site_arg(this_site));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::gather_there is deprecated, use "
+        "hpx::collectives::gather_there instead")
+    hpx::future<void> gather_there(hpx::collectives::communicator comm,
+        hpx::future<T>&& local_result, std::size_t this_site = std::size_t(-1))
+    {
+        return local_result.then([=](hpx::future<T>&& f) {
+            return hpx::collectives::gather_there(std::move(comm), f.get(),
+                hpx::collectives::this_site_arg(this_site));
+        });
+    }
+
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::create_gatherer is deprecated, use "
+        "hpx::collectives::create_communicator instead")
+    inline hpx::collectives::communicator create_gatherer(char const* basename,
+        std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::create_communicator(basename,
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation));
+    }
+}}    // namespace hpx::lcos
+
 ///////////////////////////////////////////////////////////////////////////////
 #define HPX_REGISTER_GATHER_DECLARATION(...) /**/
 

--- a/libs/full/collectives/include/hpx/collectives/inclusive_scan.hpp
+++ b/libs/full/collectives/include/hpx/collectives/inclusive_scan.hpp
@@ -14,8 +14,8 @@ namespace hpx { namespace collectives {
 
     /// Inclusive inclusive_scan a set of values from different call sites
     ///
-    /// This function receives a set of values from all call sites operating on
-    /// the given base name.
+    /// This function performs an inclusive scan operation on a set of values
+    /// received from all call sites operating on the given base name.
     ///
     /// \param  basename    The base name identifying the inclusive_scan operation
     /// \param  local_result The value to transmit to all
@@ -49,8 +49,8 @@ namespace hpx { namespace collectives {
 
     /// Inclusive inclusive_scan a set of values from different call sites
     ///
-    /// This function receives a set of values from all call sites operating on
-    /// the given base name.
+    /// This function performs an inclusive scan operation on a set of values
+    /// received from all call sites operating on the given base name.
     ///
     /// \param  comm        A communicator object returned from \a create_reducer
     /// \param  local_result The value to transmit to all

--- a/libs/full/collectives/include/hpx/collectives/scatter.hpp
+++ b/libs/full/collectives/include/hpx/collectives/scatter.hpp
@@ -323,6 +323,109 @@ namespace hpx { namespace collectives {
     }
 }}    // namespace hpx::collectives
 
+////////////////////////////////////////////////////////////////////////////////
+// compatibility functions
+namespace hpx { namespace lcos {
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::scatter_to is deprecated, use "
+        "hpx::collectives::scatter_to instead")
+    hpx::future<T> scatter_to(char const* basename,
+        std::vector<T>&& local_result, std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::scatter_to(basename, std::move(local_result),
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::scatter_to is deprecated, use "
+        "hpx::collectives::scatter_to instead")
+    hpx::future<T> scatter_to(char const* basename,
+        hpx::future<std::vector<T>>&& local_result,
+        std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1))
+    {
+        return local_result.then([=](hpx::future<T>&& f) {
+            return hpx::collectives::scatter_to(basename, f.get(),
+                hpx::collectives::num_sites_arg(num_sites),
+                hpx::collectives::this_site_arg(this_site),
+                hpx::collectives::generation_arg(generation));
+        });
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::scatter is deprecated, use hpx::collectives::scatter "
+        "instead")
+    hpx::future<T> scatter_to(hpx::collectives::communicator comm,
+        std::vector<T>&& local_result, std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::scatter_to(std::move(comm),
+            std::move(local_result),
+            hpx::collectives::this_site_arg(this_site));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::scatter_to is deprecated, use "
+        "hpx::collectives::scatter_to instead")
+    hpx::future<T> scatter_to(hpx::collectives::communicator comm,
+        hpx::future<std::vector<T>>&& local_result,
+        std::size_t this_site = std::size_t(-1))
+    {
+        return local_result.then([=](hpx::future<T>&& f) mutable {
+            hpx::collectives::scatter_to(std::move(comm), f.get(),
+                hpx::collectives::this_site_arg(this_site));
+        });
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::scatter_from is deprecated, use "
+        "hpx::collectives::scatter_from instead")
+    hpx::future<T> scatter_from(char const* basename,
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0)
+    {
+        return hpx::collectives::scatter_from<T>(basename,
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation),
+            hpx::collectives::root_site_arg(root_site));
+    }
+
+    template <typename T>
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::scatter_from is deprecated, use "
+        "hpx::collectives::scatter_from instead")
+    hpx::future<T> scatter_from(hpx::collectives::communicator comm,
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::scatter_from<T>(
+            std::move(comm), hpx::collectives::this_site_arg(this_site));
+    }
+
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::lcos::create_scatter is deprecated, use "
+        "hpx::collectives::create_communicator instead")
+    inline hpx::collectives::communicator create_scatterer(char const* basename,
+        std::size_t num_sites = std::size_t(-1),
+        std::size_t generation = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1))
+    {
+        return hpx::collectives::create_communicator(basename,
+            hpx::collectives::num_sites_arg(num_sites),
+            hpx::collectives::this_site_arg(this_site),
+            hpx::collectives::generation_arg(generation));
+    }
+}}    // namespace hpx::lcos
+
 ///////////////////////////////////////////////////////////////////////////////
 #define HPX_REGISTER_SCATTER_DECLARATION(...) /**/
 

